packages/javascript/bh-shared-ui/src/components/FileDrop/FileDrop.tsx

```diff
@@ -54,6 +56,8 @@ const FileDrop: React.FC<{
     const handleMouseEnter = (_e: MouseEvent) => setHoverActive(true);
     const handleMouseLeave = (_e: MouseEvent) => setHoverActive(false);
 
+    const formatAcceptList = () => (accept && accept.length ? accept.join(',') : undefined);
```

Comment:
This had me wondering what passing `undefined` to the input tag's accept attribute does. Do we have any plans to stop supporting `application/json` in the future? It would reduce some mental overhead for me to set a default value for the accept prop on this component of `application/json` and then only have to call `accept.join(',')` without the need to possibly set the accept list as `undefined`.

Reply:
I tested this out, passing undefined to an html prop in JSX essentially makes the output html not have that prop. I definitely can see the argument for having a default file type -- my thought process was that if we do not return a list of file types from the API endpoint, I'd rather allow the user to upload invalid types than prevent them from uploading valid types. Since we do additional validation once the file hits our API anyways

Reply:
I understand passing undefined is valid since trying this out but it took a bit of time to recognize that so my comment is more about clarity than validity. Your explanation about prevention of uploading valid file types by having a default of application/json makes sense since then one may not be able to pass a zip if something goes awry with the response for accepted types. Setting a default to an empty string still sits a little better with me since it avoids the ternary involved with undefined and will still allow all file types. Either way works though, feel free to leave as is and thanks for sharing your reasoning.

Reply:
Heyyo! Tagging on to this convo! 

I like the idea that if we don't get a list of accepted files from the API, then we default to allowing invalid types because the API is validating the content. 

I wanted to point out that In the off chance the api fails or doesnt have the list of file types, the file input would still be blocked from uploading because we check the file type before appending files to `filesForIngest`:
```ts
 const handleFileDrop = (files: FileList | null) => {
        if (files && files.length > 0) {
            const validatedFiles: FileForIngest[] = [...files].map((file) => {
                if (listFileTypesForIngest.data?.data.includes(file.type)) { // this check fails if accepted-types request fails
                    return { file, status: FileStatus.READY };
                } else {
                    return { file, errors: ['invalid file type'], status: FileStatus.READY };
                }
            });
            handleAppendFiles(validatedFiles);
        }
    };
```

I think we would need to update logic in `handleFileDrop` to allow uploading invalid types when we are missing the white list from the API!